src/System.Runtime/tests/System/Text/StringBuilderTests.netcoreapp.cs

```diff
@@ -199,5 +199,84 @@ public static void Insert_CharSpan_Invalid()
+        {
+            string mediumString = new string('a', 30);
+            string largeString = new string('b', 1000);
+
+            var sb1 = new StringBuilder("Hello");
+            var sb2 = new StringBuilder("one");
+            var sb3 = new StringBuilder(20).Append(mediumString);
+
+            yield return new object[] { new StringBuilder("Hello"), sb1, 0, 5, "HelloHello" };
+            yield return new object[] { new StringBuilder("Hello"), sb1, 0, 0, "Hello" };
+            yield return new object[] { new StringBuilder("Hello"), sb1, 2, 3, "Hellollo" };
+            yield return new object[] { new StringBuilder("Hello"), sb1, 2, 2, "Helloll" };
+            yield return new object[] { new StringBuilder("Hello"), sb1, 2, 0, "Hello" };
+            yield return new object[] { new StringBuilder("Hello"), new StringBuilder(), 0, 0, "Hello" };
+            yield return new object[] { new StringBuilder("Hello"), sb2, 0, 3, "Helloone" };
+
+            yield return new object[] { new StringBuilder("one"), sb3, 5, 25, "one" + new string('a', 25) };
+            yield return new object[] { new StringBuilder("one"), sb3, 5, 20, "one" + new string('a', 20) };
+            yield return new object[] { new StringBuilder("one"), sb3, 10, 10, "one" + new string('a', 10) };
+
+            yield return new object[] { new StringBuilder(20).Append(mediumString), sb3, 20, 10, new string('a', 40) };
+            yield return new object[] { new StringBuilder(10).Append(mediumString), sb3, 10, 10, new string('a', 40) };
+
+            yield return new object[] { new StringBuilder(20).Append(largeString), new StringBuilder(20).Append(largeString), 100, 50, largeString + new string('b', 50) };
+            yield return new object[] { new StringBuilder(10).Append(mediumString), new StringBuilder(20).Append(largeString), 20, 10, mediumString + new string('b', 10) };
+            yield return new object[] { new StringBuilder(10).Append(mediumString), new StringBuilder(20).Append(largeString), 100, 50, mediumString + new string('b', 50) };
+        }
+
+        [Theory]
+        [MemberData(nameof(Append_StringBuilder_Substring_TestData))]
+        public static void Append_StringBuilder_Substring(StringBuilder s1, StringBuilder s2, int startIndex, int count, string s)
+        {
+            Assert.Equal(s, s1.Append(s2, startIndex, count).ToString());
+        }
+
+        [Fact]
+        public static void Append_StringBuilder_InvalidInput()
+        {
+            StringBuilder sb = new StringBuilder(5, 5).Append("Hello");
+
+            Assert.Throws<ArgumentOutOfRangeException>(() => sb.Append(sb, -1, 0));
+            Assert.Throws<ArgumentOutOfRangeException>(() => sb.Append(sb, 0, -1));
+            Assert.Throws<ArgumentOutOfRangeException>(() => sb.Append(sb, 4, 5));
+            
+            Assert.Throws<ArgumentNullException>(() => sb.Append( (StringBuilder)null, 2, 2));
```

Comment:
What about `Append(null,0,1)` and `Append(null,1,0)`?

Reply:
Only ```Append(null, 0 , 0)``` return ```this``` . For all others if count and startIndex are non-negative ,the behavior will be similar to the this test